lokimq/lokimq.h

```diff
@@ -793,6 +816,31 @@ class LokiMQ {
      */
     void add_command_alias(std::string from, std::string to);
 
+    /** Creates a "tagged thread" and starts it immediately.  A tagged thread is one that batches,
+     * jobs, and timer jobs can be sent to by specifically, typically to perform coordination of
+     * some thread-unsafe work.
+     *
+     * Tagged threads will *only* process jobs sent specifically to them; they do not participate in
+     * the thread pool used for regular jobs.  Each tagged thread also has its own job queue
+     * completely separate from any other jobs.
+     *
+     * Tagged threads must be created *before* `start()` is called.  The name will be used to set the
+     * thread name in the process table (if supported on the OS).
+     *
+     * \param name - the name of the thread; will be used in log messages and (if supported by the
+     * OS) as the system thread name.
+     * \param init - an optional callback to invoke from the thread immediately after creating it.
+     * This can be used to perform initialization, however should not depend on LokiMQ having
+     * started.
+     *
+     * \param start - similar to init, but this is called immediately *after* the LokiMQ object has
+     * started up and so can use LokiMQ object functionality.
+     *
+     * \returns a TaggedThread object that can be passed to job(), batch(), or add_timer() to direct
+     * the task to the tagged thread.
+     */
+    TaggedThread add_tagged_thread(std::string name, std::function<void()> init = nullptr, std::function<void()> start = nullptr);
```

Comment:
would it also make sense to have a `destroy` function too? if so maybe a RAII type for callbacks would be better?

Reply:
The abstract class could allow that in the future, but currently the threads are static and live for the lifetime of the LokiMQ object as it keeps things simpler and is probably enough for the current use case of these threads.  (Dynamic thread creation introduces some tricky synchronization issues that are preferably avoided by just not supporting it).